src/waveform/renderers/allshader/waveformrendererlrrgb.cpp

```diff
@@ -140,48 +139,61 @@ void WaveformRendererLRRGB::paintGL() {
         const float fpos = static_cast<float>(pos);
 
         for (int chn = 0; chn < 2; chn++) {
-            float maxLow{};
-            float maxMid{};
-            float maxHigh{};
-            float maxAll{};
-
+            // Find the max values for low, mid, high and all in the waveform data
+            uchar u8maxLow{};
+            uchar u8maxMid{};
+            uchar u8maxHigh{};
+            uchar u8maxAll{};
             // data is interleaved left / right
             for (int i = visualIndexStart + chn; i < visualIndexStop + chn; i += 2) {
                 const WaveformData& waveformData = data[i];
 
-                const float filteredLow = static_cast<float>(waveformData.filtered.low);
-                const float filteredMid = static_cast<float>(waveformData.filtered.mid);
-                const float filteredHigh = static_cast<float>(waveformData.filtered.high);
+                u8maxLow = math_max(u8maxLow, waveformData.filtered.low);
+                u8maxMid = math_max(u8maxMid, waveformData.filtered.mid);
+                u8maxHigh = math_max(u8maxHigh, waveformData.filtered.high);
+                u8maxAll = math_max(u8maxAll, waveformData.filtered.all);
+            }
 
-                maxLow = math_max(maxLow, filteredLow);
-                maxMid = math_max(maxMid, filteredMid);
-                maxHigh = math_max(maxHigh, filteredHigh);
+            // Cast to float
+            float maxLow = static_cast<float>(u8maxLow);
```

Comment:
Not sure I understand why maxLow is a different type than u8maxLow. are we casting from uchar to float because of math max?

Reply:
yes, this way we cast only once, rather than casting for every iteration in the for loop in line 148.